util/execdetails/execdetails.go

```diff
@@ -47,6 +47,19 @@ type ExecDetails struct {
 	CommitDetail     *CommitDetails
 }
 
+type stmtExecDetailKeyType struct{}
+
+// StmtExecDetailKey used to carry StmtExecDetail info in context.Context.
+var StmtExecDetailKey = stmtExecDetailKeyType{}
+
+// StmtExecDetails contains stmt level execution detail info.
+type StmtExecDetails struct {
```

Comment:
How about record the Backoff by backoff_type?
And record the RPC request type too?

Such as below:
```shell
# Backoff_total_time: 10 Backoff_regionmiss_time: 8 Backoff_noleader_time: 2
# TIKV_rpc_total_time: 20 TiKV_rpc_cop_time: 15 TiKV_rpc_resolveLock_time: 5
```

Reply:
emmm..  I also think this is better to record fine-grained time，but it should introduce more performance cost, how about record total at this pr, and do fine-grained time if need in future? :D